docs/iris/src/userguide/real_and_lazy_data.rst

```diff
@@ -1,246 +1,211 @@
+   real points and bounds then the derived coordinate will also have real points
+   and bounds, otherwise the derived coordinate will have lazy points and bounds.
 
-You can use :ref:`cube maths <cube_maths>` to make in-place modifications to
-each point in a cube's existing data array.  Provided you do not directly
-reference the cube's data, the array will remain lazy:
+Iris cubes and coordinates have very similar interfaces, which extends to accessing
+coordinates' lazy points and bounds:
 
 .. doctest::
 
-    >>> cube = iris.load_cube(iris.sample_data_path('air_temp.pp'))
-    >>> cube.has_lazy_data()
+    >>> cube = iris.load_cube(iris.sample_data_path('hybrid_height.nc'))
+    >>> dim_coord = cube.coord('model_level_number')
+    >>> print dim_coord.has_lazy_points()
+    False
+    >>> print dim_coord.has_bounds()
+    False
+    >>> print dim_coord.has_lazy_bounds()
+    False
+    >>> aux_coord = cube.coord('sigma')
+    >>> print aux_coord.has_lazy_points()
     True
-    >>> cube *= 10
-    >>> cube.has_lazy_data()
+    >>> print aux_coord.has_bounds()
     True
-
-Copy
-^^^^
-
-You can copy a cube and assign a completely new data array to the copy. All the
-original cube's metadata will be the same as the new cube's metadata.  However,
-the new cube's data array will not be lazy if you replace it with a real array:
-
-.. doctest::
-
-    >>> import numpy as np
-    >>> data = np.zeros((73, 96))
-    >>> new_cube = cube.copy(data=data)
-    >>> new_cube.has_lazy_data()
+    >>> print aux_coord.has_lazy_bounds()
     False
-
-Replace
-^^^^^^^
-
-This does essentially the same thing as `cube.copy()`, except that it provides
-a safe method of doing so for the specific edge case of a lazy masked integer
-array:
-
-.. doctest::
-
-    >>> values = np.zeros((73, 96), dtype=int)
-    >>> data =np.ma.masked_values(values, 0)
-    >>> print(data)
-    [[-- -- -- ..., -- -- --]
-     [-- -- -- ..., -- -- --]
-     [-- -- -- ..., -- -- --]
-     ...,
-     [-- -- -- ..., -- -- --]
-     [-- -- -- ..., -- -- --]
-     [-- -- -- ..., -- -- --]]
-    >>> new_cube = cube.copy(data=data)
-    >>> new_cube.has_lazy_data()
+    >>> points = aux_coord.points
+    >>> print aux_coord.has_lazy_points()
     False
-    >>> new_cube.data = new_cube.lazy_data()
-    >>> new_cube.has_lazy_data()
+    >>> print derived_coord.has_lazy_points()
     True
+    >>> print derived_coord.has_bounds()
+    True
+    >>> print derived_coord.has_lazy_bounds()
+    False
 
-This method is necessary as Dask is currently unable to handle masked arrays.
-Please refer to the Whitepaper for further details.
-
-
-Coordinate Arrays
------------------
-
-Cubes possess coordinate arrays as well as data arrays, so these also benefit
-from Dask's functionality, although there are some distinctions between how
-the different coordinate types are treated.
+.. note::
+    Printing a lazy :class:`~iris.coords.AuxCoord` will realise its points and bounds arrays!
```

Comment:
Nice addition.  Sometimes we should be explicit about stuff which may seem obvious to us.